Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -577,6 +576,12 @@ public void Inline(Program program)
         PrintBplFile(Options.PrintFile, processedProgram.Program, true, true, Options.PrettyPrint);
       }
 
+      if (Options.LeanFile is not null) {
```

Comment:
I think adding more a more features to existing methods, like what happens here, makes the code less and less readable. Consider making this pipeline pluggable, so you can write the new code somewhere else.

For example:

Instead of:
```csharp
var processedProgram = PreProcessProgramVerification(program);
if (!Options.Verify)
{
  return PipelineOutcome.Done;
}

if (Options.ContractInfer)
{
  return await RunHoudini(program, stats, er);
}

if (Options.LeanFile is not null) {
  var writer = new StreamWriter(Options.LeanFile);
  LeanGenerator.EmitPassiveProgramAsLean(Options, processedProgram.Program, writer);
  writer.Close();
  return PipelineOutcome.Cancelled;
}
```

write

```csharp
var processedProgram = PreProcessProgramVerification(program);
if (!Options.Verify)
{
  return PipelineOutcome.Done;
}

if (Options.UsePassiveProgram != null && Options.UsePassiveProgram(processedProgram))
{
  return;
}
```

Note that the Houdini code has also moved to a different place.

Reply:
Yeah, I think you're right. I'll restructure it a bit.